capplets/appearance/mate-wp-xml.c

```diff
@@ -529,12 +526,17 @@ void mate_wp_xml_save_list(AppearanceData* data)
 		mate_wp_item_free(wpitem);
 	}
 
-	/* Guardamos el archivo, solo si hay nodos en <wallpapers> */
+	/* save the xml document, only if there are nodes in <wallpapers> */
 	if (xmlChildElementCount(root) > 0)
 	{
+		char *wpdir = g_build_path (G_DIR_SEPARATOR_S, g_get_user_config_dir(), "mate", NULL);
```

Comment:
I think you should use `g_build_filename()`, as it's the same but slightly simpler -- and you use it below.
```suggestion
		char *wpdir = g_build_filename (g_get_user_config_dir(), "mate", NULL);
```

Reply:
Ok, I will change it ASAP.

REF:
https://developer.gnome.org/glib/stable/glib-Miscellaneous-Utility-Functions.html#g-build-path